packages/core/src/shared/plugins/suggestion/components/SuggestionGroupItem.tsx

```diff
@@ -0,0 +1,90 @@
+import SuggestionItem from "../SuggestionItem";
+import { useEffect, useRef } from "react";
+import { Badge, createStyles, Menu, Stack, Text } from "@mantine/core";
+
+const MIN_LEFT_MARGIN = 5;
+
+export type SuggestionGroupItemProps<T> = {
+  item: T;
+  index: number;
+  selectedIndex?: number;
+  clickItem: (item: T) => void;
+};
+
+export function SuggestionGroupItem<T extends SuggestionItem>(
+  props: SuggestionGroupItemProps<T>
+) {
+  const itemRef = useRef<HTMLButtonElement>(null);
+  const { classes } = createStyles({ root: {} })(undefined, {
+    name: "SuggestionListItem",
+  });
+
+  function isSelected() {
+    const isKeyboardSelected =
+      props.selectedIndex !== undefined && props.selectedIndex === props.index;
+    const isMouseSelected = itemRef.current?.matches(":hover");
+
+    return isKeyboardSelected || isMouseSelected;
+  }
+
+  function updateSelection() {
+    isSelected()
+      ? itemRef.current?.setAttribute("data-hovered", "true")
+      : itemRef.current?.removeAttribute("data-hovered");
+  }
+
+  useEffect(() => {
+    // Updates whether the item is selected with the keyboard (triggered on selectedIndex prop change).
+    updateSelection();
+
+    if (
+      isSelected() &&
+      itemRef.current &&
+      itemRef.current.getBoundingClientRect().left > MIN_LEFT_MARGIN //TODO: Kinda hacky, fix
+      // This check is needed because initially the menu is initialized somewhere above outside the screen (with left = 1)
+      // scrollIntoView() is called before the menu is set in the right place, and without the check would scroll to the top of the page every time
+    ) {
+      itemRef.current.scrollIntoView({
+        behavior: "smooth",
+        block: "nearest",
+      });
+    }
+  });
+
+  const Icon = props.item.icon;
+
+  return (
+    <Menu.Item
+      className={classes.root}
+      icon={Icon && <Icon size={18} />}
+      onClick={() => {
+        setTimeout(() => {
```

Comment:
why do all these need a setTimeout?

Reply:
I honestly have no idea, but for whatever reason it doesn't work properly without, even though the timeout is set to 0.

Reply:
it works here without it, it seems? (chrome on mac). Are you sure?

Reply:
Oh you're right, I was looking at the onMouseLeave callback, which for sure does need it (otherwise if you hover the mouse over the item selected using the keyboard and then move the mouse elsewhere, the keyboard selection disappears). I'll remove the setTimeout from the onClick callback though👍.